### PR TITLE
Update methods to allow user status value

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: sdk/iot/hub/inc/az_iot_hub_client.h [hunk 1]
```diff
@@ -424,7 +424,7 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_parse_received_topic(
  * @param[in] client The #az_iot_hub_client to use for this call.
  * @param[in] request_id The request id. Must match a received #az_iot_hub_client_method_request
  *                       request_id.
- * @param[in] status The status. Must be 200 for "OK", 404 for "error", or 504 for "timeout"
+ * @param[in] status The status.
```
Comment (ewertons) [inline sdk/iot/hub/inc/az_iot_hub_client.h:427]: Sugg: A code that indicates the result of the method, as defined by the user.

File: sdk/iot/hub/src/az_iot_hub_client_methods.c [hunk 2]
```diff
@@ -114,15 +112,14 @@ AZ_NODISCARD az_result az_iot_hub_client_methods_response_get_publish_topic(
...
-  _az_PRECONDITION(status == 200 || status == 404 || status == 504);
   _az_PRECONDITION_NOT_NULL(mqtt_topic);
   _az_PRECONDITION(mqtt_topic_size);
 
   (void)client;
 
   az_span mqtt_topic_span = az_span_init((uint8_t*)mqtt_topic, (int32_t)mqtt_topic_size);
   int32_t required_length = az_span_size(methods_topic_prefix)
-      + az_span_size(methods_response_topic_result) + STATUS_TO_STR_SIZE
+      + az_span_size(methods_response_topic_result) + _az_iot_u32toa_size(status)
```
Comment (ewertons) [inline sdk/iot/hub/src/az_iot_hub_client_methods.c:122]: I knew that function would be useful.

File: sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c [hunk 3]
```diff
@@ -273,6 +274,51 @@ static void test_az_iot_hub_client_methods_response_get_publish_topic_succeed(vo
...
+static void test_az_iot_hub_client_methods_response_get_publish_topic_user_status_small_buf_fail(
+    void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, NULL) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_LITERAL_FROM_STR("2");
+  uint16_t status = 12345;
```
Comment (ewertons) [inline sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c:310]: To properly test the range I would suggest changing this to UINT16_MAX (65535).

File: sdk/iot/hub/inc/az_iot_hub_client.h (hunk 1 above)
Comment (danewalton) [reply sdk/iot/hub/inc/az_iot_hub_client.h:427]: updated

File: sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c (hunk 3 above)
Comment (danewalton) [reply sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c]: updated

**Pull request merged**